src/App.tsx

```diff
@@ -15,71 +16,114 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string;
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  render() {
+    const setValue = (newValue:string) => {
+      this.setState({ selectedGood: newValue });
+    };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+    const removeGood = () => {
+      this.setState({ selectedGood: '' });
+    };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        {(this.state.selectedGood === '')
+          ? (
+            <h1 className="title is-flex is-align-items-center">
+              No goods selected
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            </h1>
+          )
+          : (
+            <h1 className="title is-flex is-align-items-center">
+              { this.state.selectedGood }
+              {' '}
+              is selected
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={() => removeGood()}
+              />
```

Comment:
Button should has separate condition for visibility.

Reply:
agree, it's better to move button in a separate component